### PR TITLE
[14.0][FW-PORT] helpdesk_mgmt: Allow portal user to close its tickets

### PR DESCRIPTION
**Pull request by ypapouin**
This PR has been split from #448

- Allow a portal user to close its ticket if the final stage is set with `portal_user_can_close`.

### CODE REVIEW

File: helpdesk_mgmt/controllers/main.py [hunk 1]
```diff
@@ -11,7 +11,9 @@
 
 
 class HelpdeskTicketController(http.Controller):
-    @http.route("/ticket/close", type="http", auth="user")
+    """ """
```
Comment (pedrobaeza) [inline helpdesk_mgmt/controllers/main.py:14]: Unneeded change

File: helpdesk_mgmt/controllers/main.py [hunk 2]
```diff
@@ -20,14 +22,22 @@ def support_ticket_close(self, **kw):
...
                 values[field_name] = field_value
-        ticket = (
-            http.request.env["helpdesk.ticket"]
-            .sudo()
-            .search([("id", "=", values["ticket_id"])])
-        )
-        ticket.stage_id = values.get("stage_id")
-
-        return werkzeug.utils.redirect("/my/ticket/" + str(ticket.id))
+        if "ticket_id" in values:
```
Comment (pedrobaeza) [inline helpdesk_mgmt/controllers/main.py:25]: Why is this needed and is it related to the change of this PR?

File: helpdesk_mgmt/views/helpdesk_ticket_templates.xml [hunk 3]
```diff
@@ -136,32 +136,34 @@
...
                             </div>
+                            <div class="col-md-6">
                             <t t-foreach="closed_stages" t-as="stage">
                                 <form
-                                    method="GET"
-                                    t-if="not ticket.closed_date"
-                                    t-attf-action="/ticket/close"
-                                    style="display:inline;"
-                                >
+                                        method="GET"
```
Comment (pedrobaeza) [inline helpdesk_mgmt/views/helpdesk_ticket_templates.xml:142]: Incorrect extra indentation (for all the rest of the diff)

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Something similar has been put on 15.0 in #383, so I think it's better to backport such feature to equal modern branches.

File: helpdesk_mgmt/controllers/main.py (hunk 2 above)
Comment (ypapouin) [reply helpdesk_mgmt/controllers/main.py:25]: What do you mean, adding the check for `ticket_id` ?

Comment (pedrobaeza) [reply helpdesk_mgmt/controllers/main.py:25]: Yes, why it was not needed and now it is? And if it's an unrelated patch, it shouldn't go here.

Comment (pedrobaeza) [reply helpdesk_mgmt/controllers/main.py:25]: Anyway, I think the 15.0 change should be backported instead.
